mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java

```diff
@@ -906,4 +917,16 @@ public boolean isEnabled(int position) {
         return true;
     }
 
+    class MyImageFromItem implements SmartImage {
+        private String itemState;
+
+        public MyImageFromItem(String itemState) {
+            this.itemState = itemState;
+        }
+
+        public Bitmap getBitmap(Context context) {
+            byte[] data = Base64.decode(itemState.substring(itemState.indexOf(",") + 1), Base64.DEFAULT);
+            return BitmapFactory.decodeByteArray(data, 0, data.length);
```

Comment:
Can this also handle svg?

Reply:
I don't find on the net what formats are supported by this Android method but I can imagine it supports BMP, GIF, JPEG and PNG. At my knowledge, SVG is vectorial and not bitmap and so probably not supported as a content for an item of type Image. It would be interesting to check how ESH defines a channel of type image.

Reply:
Image item is defined in ESH as "Holds the binary data of an image". Not very precise on what formats have to be supported.
https://www.eclipse.org/smarthome/documentation/concepts/items.html

During my tests, I certainly succeeded in displaying JPEG and PNG formats in openHAB app

Reply:
I could enhance the current code to try to decode a SVG in case BitmapFactory.decodeByteArray returns null. We already have a method that build a bitmap from a SVG stream, I could reuse it.

Do you know any binding that uses SVG format with channels of type image ???
Without any binding of this kind, it will be very difficult for me to test that.

Reply:
No, it was just an idea that someone could want to use it.

Reply:
In this case, this could be implemented later when really required.